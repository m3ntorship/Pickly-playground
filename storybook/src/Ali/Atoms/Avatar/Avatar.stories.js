import React from "react";
import Avatar from "./Avatar";

export default {
  title: "Tramsi/Atoms/Avatar/Avatar",
  component: Avatar,
  argTypes: {
    Avatarsize: { control: { type: "inline-radio" } },
    Userstatus: { control: { type: "inline-radio" } },
  },
};

const Template = (args) => <Avatar {...args} />;

export const avatar = Template.bind({});
avatar.args = {
  Avatarsize: "Medium",
  Userstatus: "Filled",
};
