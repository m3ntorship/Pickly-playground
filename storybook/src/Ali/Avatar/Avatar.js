// import React from "react";
// import PropTypes from "prop-types";

// const Button = (props) => {
//   const { disabled, size, variant, children } = props;

//   return (
//     <button
//       disabled={disabled}
//       className={`capitalize rounded-full font-bold focus:outline-none disabled:opacity-25 ${
//         variant === "Primary"
//           ? `bg-primary-shd1  text-white ${
//               !disabled && "hover:bg-primary"
//             } focus:border-primary focus:border focus:ring-1 focus:ring-primary-shd1`
//           : `border text-primary-shd1 border-primary-shd1 ${
//               !disabled && "hover:bg-grey-bg"
//             } focus:ring-1 focus:ring-primary-shd1`
//       } ${
//         size === "Large"
//           ? "text-md px-xl py-s "
//           : size === "Medium"
//           ? "text-base px-l py-xs "
//           : "text-sm px-m py-xxs "
//       }
//       ${disabled ? "cursor-not-allowed" : ""}`}
//     >
//       {children}
//     </button>
//   );
// };

// Button.propTypes = {
//   size: PropTypes.oneOf(["Small", "Medium", "Large"]),
//   variant: PropTypes.oneOf(["Primary", "Secondary"]),
//   disabled: PropTypes.bool,
//   children: PropTypes.string,
// };
// export default Button;
import React from "react";
import PropTypes from "prop-types";

const Avatar = (props) => {
  const { Avatarsize, Userstatus } = props;

  return (
    <div
      className={`${
        Avatarsize === "Small"
          ? "w-8 h-8"
          : Avatarsize === "Medium"
          ? "w-10 h-10"
          : "w-14 h-14"
      } `}
    >
      <img
        className="rounded-full object-contain"
        src={`${
          Userstatus === "Filled"
            ? "/avatar.png"
            : Userstatus === "Anonymous"
            ? "/anon.svg"
            : "/default.svg"
        }`}
        alt=""
      />
    </div>
  );
};

Avatar.propTypes = {
  Avatarsize: PropTypes.oneOf(["Small", "Medium", "Large"]),
  Userstatus: PropTypes.oneOf(["Anonymous", "Filled", "Not filled"]),
};
export default Avatar;
