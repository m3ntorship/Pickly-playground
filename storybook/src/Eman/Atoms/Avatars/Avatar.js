import React from "react";
import PropTypes from "prop-types";
import Anonymous from "./../../assets/icons/Avatar/Anonymous";
import NotFilled from "./../../assets/icons/Avatar/NotFilled";
import img1 from "./../../assets/imgs/img.jfif";
import img2 from "./../../assets/imgs/img2.jfif";
import img3 from "./../../assets/imgs/img3.jpg";

export const AVATER = {
  SIZE: {
    LARGE: "Large",
    MEDIUM: "Medium",
    SMALL: "Small",
  },
  TYPE: {
    FILLED: "Filled",
    ANONYMOUS: "Anonymous",
    NOTFILLED: "Not Filled",
  },
};

export const Avater = ({ type, size, imgName }) => {
  const bgStyle =
    type === AVATER.TYPE.ANONYMOUS ? "bg-warning-shd7" : "bg-grey-shd6";
  const divSizeStyle =
    size === "Large"
      ? "w-14 h-14"
      : size === "Medium"
      ? "w-10 h-10"
      : "w-8 h-8";

  const svgSizeStyle =
    size === AVATER.SIZE.LARGE
      ? "32"
      : size === AVATER.SIZE.MEDIUM
      ? "24"
      : "16";

  return (
    <>
      <div
        className={`flex justify-center items-center rounded-full mr-4 ${divSizeStyle} ${bgStyle}`}
      >
        {type === AVATER.TYPE.FILLED && (
          <img
            className="rounded-full w-full h-full"
            src={
              imgName === "" ? img1 : `static/media/${imgName}` || img2 || img3
            } /*** At storybook imgName should be img2.e3c2ff2f.jfif or img3.c3bd2696.jpg or img.dd659847.jfif  ***/
            alt="Avater"
          />
        )}
        {type === AVATER.TYPE.ANONYMOUS && (
          <Anonymous width={svgSizeStyle} height={svgSizeStyle} />
        )}
        {type === AVATER.TYPE.NOTFILLED && (
          <NotFilled width={svgSizeStyle} height={svgSizeStyle} />
        )}
      </div>
    </>
  );
};

Avater.propTypes = {
  size: PropTypes.oneOf(["Large", "Medium", "Small"]),
  type: PropTypes.oneOf(["Filled", "Anonymous", "Not Filled"]),
  imgName: PropTypes.string,
};
