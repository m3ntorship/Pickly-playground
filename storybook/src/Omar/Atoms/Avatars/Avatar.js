import React from "react";
export const AVATAR_OPTIONS = {
  SIZE: {
    SMALL: "sm",
    MEDIUM: "md",
    BIG: "lg",
  },
};

export const Avatar = ({ anonymous, size }) => {
  return (
    <img
      src={anonymous ? "./anonymous.svg" : "https://source.unsplash.com/random"}
      alt="Profile Picture"
      className={`rounded-full object-cover mr-5 ${
        size === AVATAR_OPTIONS.SIZE.BIG
          ? "w-14 h-14"
          : size === AVATAR_OPTIONS.SIZE.MEDIUM
          ? "w-10 h-10"
          : "w-8 h-8"
      }`}
    />
  );
};
