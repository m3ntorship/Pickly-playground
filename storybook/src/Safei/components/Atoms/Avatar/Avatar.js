import React from "react";
//props to be added later to vary in size lg, md, sm based on input.
const Avatar = () => {
  return (
    <div className="inline-block">
      <img
        src="../../../../imgs/avatar.png"
        className="rounded-full h-14 w-14 mb-1"
      />
    </div>
  );
};

export default Avatar;
