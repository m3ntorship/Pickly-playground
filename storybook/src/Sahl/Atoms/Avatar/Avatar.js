import React from "react";
import PropTypes from "prop-types";

export default function Avatar(props) {
  const { size = "medium", type = "default", src, className } = props;
  let avatarPhoto;
  let imageSize = "w-10 h-10";

  // Change size
  switch (size) {
    case "small":
      imageSize = "w-6 h-6";
      break;
    case "large":
      imageSize = "w-14 h-14";
      break;
    default:
      break;
  }
  // Set the children
  switch (type) {
    case "default":
      avatarPhoto = (
        <svg
          className={`${imageSize} rounded-full bg-primary-200`}
          xmlns="http://www.w3.org/2000/svg"
          viewBox="0 0 512 512"
        >
          <path
            d="M418.88 344.021l-54.869-19.093a10.665 10.665 0 00-11.349 2.859 131.586 131.586 0 01-96.661 45.547 131.583 131.583 0 01-96.64-45.525 10.666 10.666 0 00-11.349-2.859l-54.869 19.093C37.383 363.423.006 415.978 0 475.008v15.659c0 5.891 4.776 10.667 10.667 10.667h490.667c5.891 0 10.667-4.776 10.667-10.667v-15.659c.005-59.032-37.364-111.597-93.121-130.987z"
            fill="#2196f3"
          />
          <path
            d="M384 141.44v-2.773c0-70.692-57.308-128-128-128s-128 57.308-128 128v2.773A56.914 56.914 0 00106.667 192c0 29.675 13.099 52.224 30.699 53.333C156.715 309.589 203.499 352 256 352s99.264-42.453 118.635-106.667c17.6-1.109 30.699-23.659 30.699-53.333A56.917 56.917 0 00384 141.44z"
            fill="#ffe082"
          />
          <g fill="#f44336">
            <path d="M245.333 223.808a10.453 10.453 0 01-10.453-8.853 46.04 46.04 0 0126.304-42.667 32.47 32.47 0 0016.149-33.621c1.027-10.755-6.86-20.307-17.615-21.333a19.569 19.569 0 00-3.718 0c-10.755-1.027-20.307 6.86-21.333 17.615a19.569 19.569 0 000 3.718c0 5.891-4.776 10.667-10.667 10.667s-10.667-4.776-10.667-10.667C212.284 116.152 229.684 97.05 252.199 96A40.835 40.835 0 01256 96c22.515-1.049 41.617 16.351 42.667 38.866a40.835 40.835 0 010 3.801 53.164 53.164 0 01-26.432 51.776c-17.067 10.389-16.363 21.163-16.32 21.611a10.816 10.816 0 01-9.344 11.669c-.411.051-.824.08-1.238.085z" />
            <circle cx="245.333" cy="256" r="10.667" />
          </g>
        </svg>
      );
      break;
    case "anonymous":
      avatarPhoto = (
        <svg
          className={`${imageSize} rounded-full`}
          xmlns="http://www.w3.org/2000/svg"
          fill="#fff"
          viewBox="0 0 600 600"
        >
          <defs />
          <defs>
            <clipPath id="b">
              <circle cx="300" cy="300" r="280" />
            </clipPath>
            <clipPath id="a">
              <rect width="100%" height="498" />
            </clipPath>
          </defs>
          <circle cx="300" cy="300" r="280" fill="#a1a1F1" clipPath="url(#a)" />
          <circle cx="300" cy="230" r="115" />
          <circle cx="300" cy="550" r="205" clipPath="url(#b)" />
        </svg>
      );
      break;
    case "personal":
      avatarPhoto = (
        <img
          className={`${imageSize} rounded-full object-cover block`}
          src={src}
          alt="Profile Pic"
        />
      );
      break;

    default:
      break;
  }

  return (
    <div className={`${imageSize} rounded-full ${className}`}>
      {avatarPhoto}
    </div>
  );
}

Avatar.propTypes = {
  size: PropTypes.oneOf(["small", "medium", "large"]),
  type: PropTypes.oneOf(["personal", "anonymous", "default"]),
  src: PropTypes.string,
};
