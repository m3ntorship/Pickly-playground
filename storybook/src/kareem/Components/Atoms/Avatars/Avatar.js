import React from "react";

function Avatar({ status, size }) {
  let avatarImage = "";
  let sizeStyle = "";

  switch (size) {
    case "large": {
      sizeStyle = "w-14 h-14";
      break;
    }
    case "medieum": {
      sizeStyle = "w-10 h-10";
      break;
    }
    case "small": {
      sizeStyle = "w-8 h-8";
      break;
    }
    default: {
      sizeStyle = "w-14 h-14";
      break;
    }
  }

  switch (status) {
    case "normal": {
      //avatarImage = "../Images/avatar.png";
      avatarImage = "./avatar.png";
      break;
    }
    case "anonymous": {
      //avatarImage = "../Images/anonymus.png";
      avatarImage = "./anonymus.png";
      break;
    }
    case "notfilled": {
      avatarImage = "./notfield.png";
      break;
    }
    default: {
      avatarImage = "./avatar.png";
      break;
    }
  }

  return (
    <div>
      <img
        title="dummyImg"
        data-testid="testidImg"
        src={avatarImage}
        alt={avatarImage}
        className={`${sizeStyle} rounded-xl`}
      />
    </div>
  );
}

export default Avatar;
