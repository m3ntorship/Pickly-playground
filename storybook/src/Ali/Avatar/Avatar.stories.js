import React from "react";
import Avatar from "./Avatar";

export default {
  title: "Tramsi/Atoms/Avatar/Avatar",
  component: Avatar,
};

const Template = (args) => <Avatar {...args} />;

export const avatar = Template.bind({});
avatar.args = {
  Avatarsize: "Medium",
  Userstatus: "Filled",
};
