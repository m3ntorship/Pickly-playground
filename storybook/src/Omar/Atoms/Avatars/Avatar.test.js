import React from "react";
import renderer from "react-test-renderer";
import { Avatar } from "./Avatar";

describe("Avatar Testing", () => {
  it("Avatar default testing with size big and anonymous state", () => {
    const tree = renderer.create(<Avatar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Avatar size medium testing", () => {
    const tree = renderer.create(<Avatar size="md" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Avatar size small testing", () => {
    const tree = renderer.create(<Avatar size="sm" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Avatar anonymous false state testing", () => {
    const tree = renderer
      .create(<Avatar anonymous="https://source.unsplash.com/random" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
