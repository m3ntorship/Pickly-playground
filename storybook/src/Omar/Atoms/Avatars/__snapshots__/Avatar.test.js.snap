// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Testing Avatar anonymous false state testing 1`] = `
<img
  alt="Profile Picture"
  className="rounded-full object-cover mr-5 w-8 h-8"
  src="./anonymous.svg"
/>
`;

exports[`Avatar Testing Avatar default testing with size big and anonymous state 1`] = `
<img
  alt="Profile Picture"
  className="rounded-full object-cover mr-5 w-8 h-8"
  src="https://source.unsplash.com/random"
/>
`;

exports[`Avatar Testing Avatar size medium testing 1`] = `
<img
  alt="Profile Picture"
  className="rounded-full object-cover mr-5 w-10 h-10"
  src="https://source.unsplash.com/random"
/>
`;

exports[`Avatar Testing Avatar size small testing 1`] = `
<img
  alt="Profile Picture"
  className="rounded-full object-cover mr-5 w-8 h-8"
  src="https://source.unsplash.com/random"
/>
`;
